File: _inc/_DEPRECATED_django/erp_prestech_backend/app/configs/chatify.py
import os

# -----------------------------------------------------------------------------
# MESSENGER / CHAT-LIKE SETTINGS
# -----------------------------------------------------------------------------

MESSENGER_NAME = os.getenv('CHATIFY_NAME', 'Messenger')

# The path/prefix used for chat routes in Laravel is `chats`. In Django, 
# you’d typically define this in your `urls.py` (e.g., `path('chats/', ...)`). 
# But you can store it here for convenience.
MESSENGER_PATH = os.getenv('CHATIFY_PATH', 'chats')

# Routes config in Chatify is a list of middleware, a route prefix, and a namespace. 
# In Django, route prefixes are handled in `urls.py`, and middleware is 
# usually applied globally or per-view. So you can store the concept of them here:
MESSENGER_ROUTES = {
    "PREFIX": os.getenv('CHATIFY_ROUTES_PREFIX', 'chats'),
    "MIDDLEWARE": os.getenv('CHATIFY_ROUTES_MIDDLEWARE'),
    "NAMESPACE": os.getenv('CHATIFY_ROUTES_NAMESPACE', 'myproject.mychatapp.views'),
}

# PUSHER CREDENTIALS
# Chatify uses Pusher for realtime communications; in Django, you’d typically 
# configure Pusher or a similar websocket library in your code. 
# We just store environment variables here:
PUSHER_CONFIG = {
    "KEY": os.getenv('PUSHER_APP_KEY'),
    "SECRET": os.getenv('PUSHER_APP_SECRET'),
    "APP_ID": os.getenv('PUSHER_APP_ID'),
    "OPTIONS": {
        "cluster": os.getenv('PUSHER_APP_CLUSTER'),
        "useTLS": os.getenv('PUSHER_APP_USETLS', 'false').lower() == 'true',
    },
}

# USER AVATAR SETTINGS
MESSENGER_USER_AVATAR = {
    "FOLDER": "uploads/avatar",
    "DEFAULT": "avatar.png",
}

# ATTACHMENTS SETTINGS
MESSENGER_ATTACHMENTS = {
    "FOLDER": "attachments",
    # In Laravel Chatify, "download_route_name" is the named route for downloads.
    # In Django, you’d define a route in your urls.py like `path('attachments/<id>/download', ...)`.
    "DOWNLOAD_ROUTE_NAME": "attachments_download",

    # Allowed file extensions
    "ALLOWED_IMAGES": ["png", "jpg", "jpeg", "gif"],
    "ALLOWED_FILES": ["zip", "rar", "txt"],
}

File: _inc/laravel/config/chatify.php
<?php

use App\Config\Constants\MiddlewaresConstants;

return [

    /*
    |-------------------------------------
    | Messenger display name
    |-------------------------------------
    */
    'name' => env('CHATIFY_NAME', 'Messenger'),


    /*
   |--------------------------------------------------------------------------
   | Package path
   |--------------------------------------------------------------------------
   |
   | This value is the path of the package or in other meaning, it is the prefix
   | of all the registered routes in this package.
   |
   | e.g. : app.test/chatify
   */

    'path' => env('CHATIFY_PATH', 'chats'),

    /*
    |-------------------------------------
    | Routes configurations
    |-------------------------------------
    */
    'routes' => [
        'prefix' => env('CHATIFY_ROUTES_PREFIX', 'chats'),
        'middleware' => env(
            'CHATIFY_ROUTES_MIDDLEWARE',
            [
                MiddlewaresConstants::WEB,
                MiddlewaresConstants::AUTH,
                MiddlewaresConstants::XSS,
                MiddlewaresConstants::PSR,
            ]
        ),
        'namespace' => env('CHATIFY_ROUTES_NAMESPACE', 'Chatify\Http\Controllers'),
    ],

    'api_routes' => [
        'prefix' => env('CHATIFY_API_ROUTES_PREFIX', 'api/chats'),
        'middleware' => env('CHATIFY_API_ROUTES_MIDDLEWARE', ['api', 'auth:sanctum']),
        'namespace' => env('CHATIFY_API_ROUTES_NAMESPACE', 'Chatify\Http\Controllers\Api'),
    ],


    /*
    |-------------------------------------
    | Pusher API credentials
    |-------------------------------------
    */
    'pusher' => [
        'key' => env('PUSHER_APP_KEY'),
        'secret' => env('PUSHER_APP_SECRET'),
        'app_id' => env('PUSHER_APP_ID'),
        'options' => (array)[
            'cluster' => env('PUSHER_APP_CLUSTER'),
            'useTLS' => env('PUSHER_APP_USETLS'),
        ],
    ],

    /*
    |-------------------------------------
    | User Avatar
    |-------------------------------------
    */
    'user_avatar' => [
        'folder' => 'uploads/avatar',
        'default' => 'avatar.png',
    ],

    /*
    |-------------------------------------
    | Attachments
    |-------------------------------------
    */
    'attachments' => [
        'folder' => 'attachments',
        'download_route_name' => 'attachments.download',
        'allowed_images' => (array)[
            'png',
            'jpg',
            'jpeg',
            'gif',
        ],
        'allowed_files' => (array)[
            'zip',
            'rar',
            'txt',
        ],
    ],
];
